src/NovelRT.Interop/Graphics/NrtRGBAColour.cpp

```diff
@@ -14,132 +14,30 @@ extern "C"
-        {
-            Nrt_setErrMsgIsNullptrInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        colourConfigPtr->setG(inputValue);
-
-        return NRT_SUCCESS;
-    }
-
-    int32_t Nrt_RGBAColour_getB(NrtRGBAColourHandle colourConfig)
-    {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        return colourConfigPtr->getB();
-    }
-
-    NrtResult Nrt_RGBAColour_setB(NrtRGBAColourHandle colourConfig, int32_t inputValue)
-    {
-        if (colourConfig == nullptr)
-        {
-            Nrt_setErrMsgIsNullptrInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        colourConfigPtr->setB(inputValue);
-
-        return NRT_SUCCESS;
-    }
-
-    int32_t Nrt_RGBAColour_getA(NrtRGBAColourHandle colourConfig)
-    {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        return colourConfigPtr->getA();
-    }
-
-    NrtResult Nrt_RGBAColour_setA(NrtRGBAColourHandle colourConfig, int32_t inputValue)
-    {
-        if (colourConfig == nullptr)
-        {
-            Nrt_setErrMsgIsNullptrInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        colourConfigPtr->setA(inputValue);
-
-        return NRT_SUCCESS;
-    }
-
-    float Nrt_RGBAColour_getRScalar(NrtRGBAColourHandle colourConfig)
-    {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
+        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(&colourConfig);
         return colourConfigPtr->getRScalar();
     }
 
-    float Nrt_RGBAColour_getGScalar(NrtRGBAColourHandle colourConfig)
+    float Nrt_RGBAColour_getGScalar(NrtRGBAColour colourConfig)
     {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
+        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(&colourConfig);
         return colourConfigPtr->getGScalar();
     }
 
-    float Nrt_RGBAColour_getBScalar(NrtRGBAColourHandle colourConfig)
+    float Nrt_RGBAColour_getBScalar(NrtRGBAColour colourConfig)
     {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
+        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(&colourConfig);
         return colourConfigPtr->getBScalar();
     }
 
-    float Nrt_RGBAColour_getAScalar(NrtRGBAColourHandle colourConfig)
+    float Nrt_RGBAColour_getAScalar(NrtRGBAColour colourConfig)
     {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
+        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(&colourConfig);
         return colourConfigPtr->getAScalar();
     }
```

Comment:
Same for these three, I think this might be more performant if we dereferenced the result of `reinterpret_cast` and used it as a value.

Reply:
I agree, there's a lot of (de)referencing in there.